### PR TITLE
Add Regex type

### LINKED ISSUE #14: Add =~ operator for regex check
I suggest to add a new infix operator `=~` for checking whether a string matches a regex using the following implementation:

```
infix operator =~
func =~ (lhs: String, rhs: String) -> Bool {
    return lhs.range(of: rhs, options: .regularExpression) == lhs.range(of: lhs)
}
```

Use:

`let stringMatchesRegex = "ABC" =~ "[A-Z]+"`

### PR DESCRIPTION
**Pull request by fredpi**
This is my suggestion for Regex in HandySwift. Please comment on whether this is suitable.

Note: It's not tested and documented yet, I will do that on approval.

Would close #14.

### CODE REVIEW

File: Sources/Structs/Regex.swift [hunk 1]
```diff
@@ -0,0 +1,232 @@
+//
+//  Regex.swift
+//  HandySwift
+//
+//  Created by Frederick Pietschmann on 19.03.18.
+//  Copyright © 2018 Flinesoft. All rights reserved.
+//
+// Taken from https://github.com/sharplet/Regex, then modified to remove some weight.
```
Comment (Dschee) [inline Sources/Structs/Regex.swift]: Use `Originally from: https://...` instead. Also place a second whitespace at the beginning to allign with the above.

File: Sources/Structs/Regex.swift [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    }
+
+    // MARK: - Methods: Matching
+    /// Returns `true` if the regex matches `string`, otherwise returns `false`.
+    ///
+    /// - parameter string: The string to test.
+    ///
+    /// - returns: `true` if the regular expression matches, otherwise `false`.
+    public func matches(_ string: String) -> Bool {
+        return !matches(in: string).isEmpty
```
Comment (Dschee) [inline Sources/Structs/Regex.swift]: I feel like this is enough reason to also introduce the `firstMatch` method and use it here. This implementation will be very slow (even in average) if the underlying text is a long text. Using `firstMatch` here would improve the average performance a lot for many cases. Please also note that the `firstMatch(in:)` should be public as I feel like it's needed as often as `matches(in:) -> [Match])` is.

**Comment by Dschee** [comment]
Any updated on this?

**Comment by fredpi** [comment]
@Dschee I added the first match interface as suggested, while documentation and tests are yet to be created.

File: Sources/Structs/Regex.swift [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+    /// If the regex matches `string`, returns a `Match` describing the
+    /// first matched string and any captures. If there are no matches, returns
+    /// `nil`.
+    ///
+    /// - parameter string: The string to match against.
+    ///
+    /// - returns: An optional `Match` describing the first match, or `nil`.
+    ///
+    /// - note: If the match is successful, the result is also stored in `Regex.lastMatch`.
```
Comment (Dschee) [inline Sources/Structs/Regex.swift]: Not sure if this is true. You removed the lastMatch, no?

**Comment by Dschee** [comment]
Nice, seems like only a new README section is missing, no? Thank you very much for all the work done so far. 👍

**Comment by fredpi** [comment]
@Dschee You're welcome. With the readme now being done, this PR is now ready to merge if there aren't any issues / further suggestions.

**Comment by Dschee** [comment]
Looks good, merging. 🎉

**Pull request merged**